saas/api/subscriptions.py

```diff
@@ -1005,7 +1005,7 @@ def subscription(self):
 
     def get_queryset(self):
         queryset = Subscription.objects.active_with(
-            self.kwargs.get(self.provider_url_kwarg))
+            self.kwargs.get(self.provider_url_kwarg), request_key=self.kwargs.get('request_key'))
```

Comment:
The issue solved here is that `active_with` calls `valid_for`, hence the returned queryset only contains `Subscription` instances where `request_key` is `None`, hence the property `SubscriptionRequestAcceptAPIView.subscription` always raises a 404 Not Found.

Better fix is to remove the `get_queryset` altogether, and add `ProvidedSubscriptionsMixin` in the list of parent classes.